src/dune_file.ml

```diff
@@ -1659,25 +1659,31 @@ module Menhir = struct
     ; modules    : string list
     ; mode       : Rule.Mode.t
     ; loc        :  Loc.t
+    ; infer      : bool
     }
 
   let syntax =
     Syntax.create
       ~name:"menhir"
       ~desc:"the menhir extension"
-      [ (1, 0) ]
+      [ 1, 0
+      ; 2, 0
+      ]
 
   let decode =
     record
       (let%map merge_into = field_o "merge_into" string
        and flags = field_oslu "flags"
        and modules = field "modules" (list string)
-       and mode = Rule.Mode.field in
+       and mode = Rule.Mode.field
+       and infer = field_b "infer" ~check:(Syntax.since syntax (2, 0))
```

Comment:
Actually, we should always revisit the default values after a major version bump, otherwise the dune files will keep growing forever. It's often the case that the right default at a certain time is the wrong default 6 months later.

Reply:
Okay, the default now depends on the version. There was no appropriate field primitive for parsing a boolean field with no default, so I introduced one.